bcipy/helpers/stimuli.py

```diff
@@ -227,6 +228,144 @@ def extract_trials(
+            labels (List[str]): Target letter in each inquiry.
+
+            data_by_targets (dict): Dictionary where keys are the symbol set and values are the appended inquiries
+            for each symbol. dict[Key] = (np.ndarray) of shape (Channels, Samples)
+        """
+        if channel_map:
+            # Remove the channels that we are not interested in
+            channels_to_remove = [idx for idx, value in enumerate(channel_map) if value == 0]
+            gaze_data = np.delete(gaze_data, channels_to_remove, axis=0)
+
+        # Find the value closest to (& greater than) inq_start_times
+        gaze_data_timing = gaze_data[-1, :].tolist()
+
+        start_times = []
+        for times in inq_start_times:
+            temp = list(filter(lambda x: x > times, gaze_data_timing))
+            if len(temp) > 0:
+                start_times.append(temp[0])
+
+        triggers = []
+        for val in start_times:
+            triggers.append(gaze_data_timing.index(val))
+
+        # Label for every inquiry
+        labels = target_symbols
+
+        symbol_set = alphabet()
+
+        # Create a dictionary with symbols as keys and data as values
+        # 'A': [], 'B': [] ...
+        data_by_targets = {}
+        for symbol in symbol_set:
+            data_by_targets[symbol] = []
+
+        reshaped_data = []
+        # Merge the inquiries if they have the same target letter:
+        for i, inquiry_index in enumerate(triggers):
+            start = inquiry_index
+            stop = int(inquiry_index + (sample_rate * 3))   # (60 samples * 3 seconds)
+            # Check if the data exists for the inquiry:
+            if stop > len(gaze_data[0, :]):
+                continue
+
+            reshaped_data.append(gaze_data[:, start:stop])
+            # (Optional) extracted data (Inquiries x Channels x Samples)
+
+            # Populate the dict by appending the inquiry to the correct key:
+            data_by_targets[labels[i]].append(gaze_data[:, start:stop])
+
+        # After populating, flatten the arrays in the dictionary to (Channels x Samples):
+        for symbol in symbol_set:
+            if len(data_by_targets[symbol]) > 0:
+                data_by_targets[symbol] = np.transpose(np.array(data_by_targets[symbol]), (1, 0, 2))
+                data_by_targets[symbol] = np.reshape(data_by_targets[symbol], (len(data_by_targets[symbol]), -1))
+
+            # TODO: Note that this is a workaround to the issue of having different number of targetness in
+            # each symbol. If a target symbol is prompted more than once, the data is appended to the dict as a list.
+            # Which is why we need to convert it to a (np.ndarray) and flatten the dimensions.
+            # This is not ideal, but it works for now.
+
+        # return np.stack(reshaped_data, 0), labels
+        return data_by_targets
+
+    @staticmethod
+    def extract_eye_info(data):
```

Comment:
This could be moved to a signal process method

Reply:
I created a top-level function `extract_gaze.py` in signal/model module. However I'm not sure I understood why we can't keep this as a static method within Gaze Reshaper. In a similar example, we have `extract_trials` https://github.com/CAMBI-tech/BciPy/blob/2.0.0rc4/bcipy/helpers/stimuli.py#L184 within Inquiry Reshaper that does the equivalent work for EEG data.